services/verification/src/services/Injector.ts

```diff
@@ -689,44 +838,31 @@ export class Injector {
     contract: CheckedContract,
     deployerAddress: string,
     salt: string,
-    constructorArgs: Create2ConstructorArgument[],
-    create2Address: string
+    create2Address: string,
+    abiEncodedConstructorArguments?: string
   ): Promise<Match> {
     if (!CheckedContract.isValid(contract)) {
       await CheckedContract.fetchMissing(contract);
     }
 
-    const constructorArgsTypes = constructorArgs.map(
-      (arg: Create2ConstructorArgument) => arg.type
-    );
-    const constructorArgsValues = constructorArgs.map(
-      (arg: Create2ConstructorArgument) => arg.value
-    );
-
     const compilationResult = await recompile(
       contract.metadata,
       contract.solidity
     );
 
-    const computedAddr = getCreate2Address({
-      factoryAddress: deployerAddress,
+    const computedAddr = getCreate2Address(
+      deployerAddress,
       salt,
-      contractBytecode: compilationResult.creationBytecode,
-      constructorTypes: constructorArgsTypes,
-      constructorArgs: constructorArgsValues,
-    });
+      compilationResult.creationBytecode,
+      abiEncodedConstructorArguments
+    );
 
-    if (create2Address !== computedAddr) {
+    if (create2Address.toLowerCase() !== computedAddr.toLowerCase()) {
```

Comment:
I don't know if we want this check... Because the users is expecting that the create2Address they pass are exactly the addresses computed by us. It would be better to do a checksum verification returning an error in the case the passed create2address is not checksummed

Reply:
Checksummed and non-cheksummed addresses are not different than each other. Bytewise `0xAbCdEf` is equivalent to `0xabcdef`. Checksum is generally for avoiding typos etc. when taking an irreversible action like sending a tx. I don't think it applies to our case.

Reply:
I know that one usecase of create2 is to have fancy addresses, so maybe the user will search for a specific address like 0xAaAaAa and she thinks she found it but then she has problems for this check, I don't know if I explained myself, kinda a specific edge case I know :D